lib/ecto/changeset.ex

```diff
@@ -1192,10 +1192,14 @@ defmodule Ecto.Changeset do
     end
   end
 
-  defp put_change(data, _changes, _errors, _valid?, key, _value, nil) do
+  defp put_change(data, _changes, _errors, _valid?, key, _value, nil) when is_atom(key) do
     raise ArgumentError, "unknown field `#{inspect(key)}` in #{inspect(data)}"
   end
 
+  defp put_change(_data, _changes, _errors, _valid?, key, _value, nil) when not is_atom(key) do
+    raise ArgumentError, "field names given to change/put_change must be atoms, got: `#{inspect(key)}`"
```

Comment:
we discussed augmenting the existing error message, however if we put it at the end of the error it can easily get lost:

<img width="1515" alt="image" src="https://user-images.githubusercontent.com/76071/66121603-da07fb00-e5dd-11e9-95d0-0e59320c900b.png">